pkg/controllers/dynakube/processmoduleconfigsecret/reconciler.go

```diff
@@ -0,0 +1,140 @@
+package processmoduleconfigsecret
+
+import (
+	"context"
+	"encoding/json"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/timeprovider"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	PullSecretSuffix             = "-pmc-secret"
+	SecretKeyProcessModuleConfig = "ruxitagentproc.conf"
+)
+
+type Reconciler struct {
+	client       client.Client
+	apiReader    client.Reader
+	dtClient     dtclient.Client
+	dynakube     *dynatracev1beta1.DynaKube
+	scheme       *runtime.Scheme
+	timeProvider *timeprovider.Provider
+}
+
+func NewReconciler(clt client.Client, //nolint:revive
+	apiReader client.Reader,
+	dtClient dtclient.Client,
+	dynakube *dynatracev1beta1.DynaKube,
+	scheme *runtime.Scheme,
+	timeProvider *timeprovider.Provider) *Reconciler {
+	return &Reconciler{
+		client:       clt,
+		apiReader:    apiReader,
+		dtClient:     dtClient,
+		dynakube:     dynakube,
+		scheme:       scheme,
+		timeProvider: timeProvider,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context) error {
+	if r.dynakube.CloudNativeFullstackMode() || r.dynakube.ApplicationMonitoringMode() {
+		err := r.reconcileSecret(ctx)
+		if err != nil {
+			log.Info("could not reconcile pull secret")
+			return errors.WithStack(err)
+		}
+	} else {
+		log.Info("skipping process module secret reconciler")
+	}
+
+	return nil
+}
+
+func (r *Reconciler) reconcileSecret(ctx context.Context) error {
+	secret, err := r.getOrCreateSecretIfNotExists(ctx)
+	if err != nil {
+		return errors.WithMessage(err, "could not get or create secret")
+	}
+
+	if err := r.updateSecretIfOutdated(ctx, secret); err != nil {
+		return errors.WithMessage(err, "could not update secret")
+	}
+	return nil
+}
+
+func (r *Reconciler) getOrCreateSecretIfNotExists(ctx context.Context) (*corev1.Secret, error) { // created, error
+	var config corev1.Secret
+	err := r.apiReader.Get(ctx, client.ObjectKey{Name: extendWithSuffix(r.dynakube.Name), Namespace: r.dynakube.Namespace}, &config)
+	if k8serrors.IsNotFound(err) {
+		log.Info("creating pull secret")
```

Comment:
is the naming pull secret correct here? I know just a minor nitpick but I am curious :slightly_smiling_face: